set.mm

```diff
@@ -714243,6 +714297,465 @@ valid cases ( ` (/) ` is the last symbol) and invalid cases ( ` (/) `
     ZABUMZVQTDZVRWCQABUDABUEWEWFRZVRWCWGVRRWDAVTBWEWFVRUFWFVRVTWDDZWEWFVRRZVTTD
     ZVQUNDZVTVQUGUHZWHWIWKWJWKWIVQUIUJZVQUKULWMWFWLVRWFVQVQUOUPUQVTVQURUSUTVAVB
     VEWCVPWAEWCWAEPZEADZVPVFWNWCWOWCWOQEWAWCEWAPWOWAAEVCSVDSEAVGVHVIVJVKVLVM $.
+
+
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Proper (unordered) pairs
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+
+  Proper (unordered) pairs are unordered pairs with exactly 2 elements.  The
+  set of proper pairs with elements of a class ` V ` is defined by
+  ` { x e. ~P V | ( # `` x ) = 2 } `.
+
+  For example, ` { 1 ,  2 } ` is a proper pair, because ` 1 =/= 2 ` ( see
+  ~ 1ne2 ).  Examples for not proper unordered pairs are ` { 1 , 1 } = { 1 } `
+  (see ~ preqsn ), ` { 1 , _V } = { 1 } ` (see ~ prprc2 ) or
+  ` { _V , _V } = (/) ` (see ~ prprc ).
+
+$)
+
+  ${
+    $d V x $.  $d V a b $.  $d X x $.  $d X a b $.
+    prpair.p $e |- P = { x e. ~P V | ( # ` x ) = 2 } $.
+    $( Characterization of a proper pair:  A class is a proper pair iff it
+       consists of exactly two different sets.  (Contributed by AV,
+       11-Mar-2023.) $)
+    prpair $p |- ( X e. P <-> E. a e. V E. b e. V
+                              ( X = { a , b } /\ a =/= b ) ) $=
+      ( wcel cv chash cfv c2 wceq cpw crab wa wrex fveqeq2 imp adantr cpr elrab
+      wne eleq2i hash2prb wss elpwi ancom 2rexbii biimpi ss2rexv syl2im prelpwi
+      sylbid hashprg biimpd adantld wb eleq1 anbi12d adantl mpbir2and rexlimivv
+      ex impbii 3bitri ) DBHDAIZJKLMZACNZOZHDVIHZDJKLMZPZDEIZFIZUAZMZVNVOUCZPZF
+      CQECQZBVJDGUDVHVLADVIVGDLJRUBVMVTVKVLVTVKVLVRVQPZFDQEDQZVTDVIEFUEVKDCUFWB
+      VSFDQEDQZVTDCUGWBWCWAVSEFDDVRVQUHUIUJVSEFDCUKULUNSVSVMEFCCVNCHVOCHPZVSVMW
+      DVSPVMVPVIHZVPJKLMZWDWEVSVNVOCUMTWDVSWFWDVRWFVQWDVRWFVNVOCCUOUPUQSVSVMWEW
+      FPURZWDVQWGVRVQVKWEVLWFDVPVIUSDVPLJRUTTVAVBVDVCVEVF $.
+  $}
+
+  ${
+    $d R x $.  $d V p $.  $d V x $.  $d W p $.  $d W x $.
+    prproropf1o.o $e |- O = { x e. ( V X. V ) | ( 1st ` x ) R ( 2nd ` x ) } $.
+    prproropf1o.p $e |- P = { p e. ~P V | ( # ` p ) = 2 } $.
+    $( Lemma 1 for ~ prproropf1o .  (Contributed by AV, 12-Mar-2023.) $)
+    prproropf1olem1 $p |- ( ( R Or V /\ W e. O )
+                            -> { ( 1st ` W ) , ( 2nd ` W ) } e. P ) $=
+      ( wcel wa c1st cfv c2nd chash c2 wceq wbr cv cvv wor cpr cpw eleq2i fveq2
+      cxp crab breq12d elrab bitri xp1st xp2nd jca prelpwi syl ad2antrl wne wpo
+      sopo adantr simprr po2ne syl3anc fvex hashprg mp2an sylib sylan2b fveqeq2
+      wb sylibr ) ECUAZFDJZKFLMZFNMZUBZEUCZJZVPOMPQZKZVPBJZVMVLFEEUFZJZVNVOCRZK
+      ZVTVMFASZLMZWFNMZCRZAWBUGZJWEDWJFHUDWIWDAFWBWFFQWGVNWHVOCWFFLUEWFFNUEUHUI
+      UJVLWEKZVRVSWCVRVLWDWCVNEJZVOEJZKZVRWCWLWMFEEUKFEEULUMZVNVOEUNUOUPWKVNVOU
+      QZVSWKECURZWNWDWPVLWQWEECUSUTWCWNVLWDWOUPVLWCWDVAVNVOCEVBVCVNTJVOTJWPVSVJ
+      FLVDFNVDVNVOTTVEVFVGUMVHWAVPGSZOMPQZGVQUGZJVTBWTVPIUDWSVSGVPVQWRVPPOVIUIU
+      JVK $.
```

Comment:
Your local definition of `O` is actually the same as ``R |` ( V X. V )``. Maybe this could lead to simplifications?

I came up with this quick proof to convince myself:
```
h1::prproropor.o   |- O = { x e. ( V X. V ) | ( 1st ` x ) R ( 2nd ` x ) }
2:1:eleq2i         |- ( x e. O <-> x e. { x e. ( V X. V ) | ( 1st ` x ) R ( 2nd ` x ) } )
3::rabid           |- ( x e. { x e. ( V X. V ) | ( 1st ` x ) R ( 2nd ` x ) } <-> ( x e. ( V X. V ) /\ ( 1st ` x ) R ( 2nd ` x ) ) )
4:2,3:bitri        |- ( x e. O <-> ( x e. ( V X. V ) /\ ( 1st ` x ) R ( 2nd ` x ) ) )
5::1st2nd2         |- ( x e. ( V X. V ) -> x = <. ( 1st ` x ) , ( 2nd ` x ) >. )
6:5:adantl         |- ( ( R C_ ( V X. V ) /\ x e. ( V X. V ) ) -> x = <. ( 1st ` x ) , ( 2nd ` x ) >. )
7:6:eleq1d         |- ( ( R C_ ( V X. V ) /\ x e. ( V X. V ) ) -> ( x e. R <-> <. ( 1st ` x ) , ( 2nd ` x ) >. e. R ) )
8::df-br           |- ( ( 1st ` x ) R ( 2nd ` x ) <-> <. ( 1st ` x ) , ( 2nd ` x ) >. e. R )
9:7,8:syl6rbbr     |- ( ( R C_ ( V X. V ) /\ x e. ( V X. V ) ) -> ( ( 1st ` x ) R ( 2nd ` x ) <-> x e. R ) )
10:9:pm5.32da      |- ( R C_ ( V X. V ) -> ( ( x e. ( V X. V ) /\ ( 1st ` x ) R ( 2nd ` x ) ) <-> ( x e. ( V X. V ) /\ x e. R ) ) )
11::ssel           |- ( R C_ ( V X. V ) -> ( x e. R -> x e. ( V X. V ) ) )
12:11:pm4.71rd     |- ( R C_ ( V X. V ) -> ( x e. R <-> ( x e. ( V X. V ) /\ x e. R ) ) )
13:10,12:bitr4d    |- ( R C_ ( V X. V ) -> ( ( x e. ( V X. V ) /\ ( 1st ` x ) R ( 2nd ` x ) ) <-> x e. R ) )
14:4,13:syl5bb     |- ( R C_ ( V X. V ) -> ( x e. O <-> x e. R ) )
qed:14:eqrdv       |- ( R C_ ( V X. V ) -> O = R )
```

Reply:
It looks more like `O = ( R i^i ( V X. V ) )`. But probably the version bypassing the definition of `O` is better.

Reply:
Of course it was `O = ( R i^i ( V X. V ) )`, thanks for noticing!

Reply:
OK, I will replace the local defintion of `O` by `O = ( R i^i ( V X. V ) )`, although it does not eliminate the local variable (of course it could, but it would make the theorems themselves more clumsy), and I think it will not simplify the proofs a lot. The version with ` R C_ ( V X. V )` in the antecedent is more restrictive, so I will not consider it further.

Reply:
Local defintion of O replaced by O = ( R i^i ( V X. V ) ). Actually, most of the proofs could have been shortened, even without using the additional lemma ~prproropf1olem0.